### PR TITLE
Academic Calendar

### PR DESCRIPTION
**Pull request by DerrickLi**
Here is the preliminary design of the academic calendar.

53781351_266924124209243_1527247184561438720_n

Some differences from Yani's design:
1. I used a TabLayout instead of buttons for code reuse reasons, but I have somewhat of an idea to do it with buttons if that design is preferable.
2. Yiling and I couldn't really find any relevant calendars besides the academic calendar, l&s dates, and final exam dates. Since there aren't too many calendars I decided to make the "all" tab show the overall academic calendar. Could change this if we end up finding enough useful calendars?

### CODE REVIEW

File: app/src/main/java/com/asuc/asucmobile/adapters/CalendarAdapter.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package com.asuc.asucmobile.adapters;
```
Comment (rustielin) [inline app/src/main/java/com/asuc/asucmobile/adapters/CalendarAdapter.java]: belongs in `domain` package

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -63,10 +62,10 @@
         <meta-data
             android:name="firebase_crashlytics_collection_enabled"
             android:value="false" />
-        <meta-data android:name="firebase_messaging_auto_init_enabled"
+        <meta-data
```
Comment (rustielin) [inline app/src/main/AndroidManifest.xml]: ??

File: app/src/main/java/com/asuc/asucmobile/domain/fragments/CalendarFragment.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+     */
+    public static CalendarFragment newInstance(int sectionNumber) {
+        CalendarFragment fragment = new CalendarFragment();
+        Bundle args = new Bundle();
+        args.putInt(ARG_SECTION_NUMBER, sectionNumber);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    private static final String[] info_calendar = new String[] {"L&S Deadlines", "Academic Calendar"};
```
Comment (rustielin) [inline app/src/main/java/com/asuc/asucmobile/domain/fragments/CalendarFragment.java]: best practice: these belong in `values` as Lombok utility classes

File: app/src/main/java/com/asuc/asucmobile/domain/fragments/PlaceholderFragment.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import android.support.v4.app.Fragment;
+
+import com.asuc.asucmobile.domain.main.CalendarActivity;
+
+/**
+ * This fragment is used to create a new intent to open the tabbed Calendar Activity because
+ * the Navigation Generator can only open fragments. REEEEEEEE
+ */
+public class PlaceholderFragment extends Fragment {
+    public PlaceholderFragment() {
```
Comment (rustielin) [inline app/src/main/java/com/asuc/asucmobile/domain/fragments/PlaceholderFragment.java]: can replace with Lombok noargs constructor

File: app/src/main/java/com/asuc/asucmobile/domain/main/CalendarActivity.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        public int getCount() {
+            // Show 3 total pages.
+            return 3;
+        }
+
+        @Override
+        public CharSequence getPageTitle(int position) {
+            switch (position) {
+                case 0:
+                    return "ALL";
```
Comment (rustielin) [inline app/src/main/java/com/asuc/asucmobile/domain/main/CalendarActivity.java]: best practice: values belong as Lombok utility class

File: app/src/main/java/com/asuc/asucmobile/models/CalendarItem.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package com.asuc.asucmobile.models;
+
+public class CalendarItem {
+    private String date;
+    private String info;
+    private boolean isHeader;
+
+    public CalendarItem(String deadline, String information, boolean header) {
```
Comment (rustielin) [inline app/src/main/java/com/asuc/asucmobile/models/CalendarItem.java]: can replace with Lombok allargs constructor

File: app/src/main/java/com/asuc/asucmobile/models/CalendarItem.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        date = deadline;
+        info = information;
+        isHeader = header;
+    }
+
+    public String getDate() {
```
Comment (rustielin) [inline app/src/main/java/com/asuc/asucmobile/models/CalendarItem.java]: can replace with lombok getter

File: app/src/main/java/com/asuc/asucmobile/domain/fragments/CalendarFragment.java (hunk 3 above)
Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/domain/fragments/CalendarFragment.java]: weren't we gonna pull this from firebase?

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (DerrickLi) [reply app/src/main/AndroidManifest.xml]: what's wrong with this ?

File: app/src/main/java/com/asuc/asucmobile/adapters/CalendarAdapter.java (hunk 1 above)
Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/adapters/CalendarAdapter.java]: addressed

File: app/src/main/java/com/asuc/asucmobile/domain/fragments/PlaceholderFragment.java (hunk 4 above)
Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/domain/fragments/PlaceholderFragment.java]: wow wtf that's cool

Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/domain/fragments/PlaceholderFragment.java]: addressed

File: app/src/main/java/com/asuc/asucmobile/domain/main/CalendarActivity.java (hunk 5 above)
Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/domain/main/CalendarActivity.java]: okay :/
addressed

File: app/src/main/java/com/asuc/asucmobile/models/CalendarItem.java (hunk 6 above)
Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/models/CalendarItem.java]: addressed

File: app/src/main/java/com/asuc/asucmobile/domain/fragments/CalendarFragment.java (hunk 3 above)
Comment (DerrickLi) [reply app/src/main/java/com/asuc/asucmobile/domain/fragments/CalendarFragment.java]: nvm addressed

**Pull request merged**